hw31/js/snake.js

```diff
@@ -0,0 +1,232 @@
+            gridCellCssClass: Snake.snakeCellCssClass,
+            gridContainerSelector: Snake.gridContainerSelector,
+        });
+
+        this.#foodImage.src = Snake.foodImage;
+
+        this.#init();
+    }
+
+
+    #init() {
+        this.#startBtn.addEventListener('click', () => {
+            this.#start();
+        });
+
+        this.#endBtn.addEventListener('click', () => {
+            this.#endGame();
+        });
+
+        document.addEventListener('keydown', (event) => {
+            this.#updateDirection(event);
+        });
+    }
+
+    #showStartGameButton() {
+        this.#startBtn.classList.remove('hidden');
+        this.#endBtn.classList.add('hidden');
+    }
+
+    #showEndGameButton() {
+        this.#startBtn.classList.add('hidden');
+        this.#endBtn.classList.remove('hidden');
+    }
+
+    #increaseScore() {
+        this.#score += 1;
+        this.#scoreValue.textContent = this.#score.toString();
+    }
+
+    #clearScore() {
+        this.#score = 0;
+    }
+
+    #removeFood() {
+        this.#food.removeChild(this.#food.firstChild);
+    }
+
+    #findByCoords({row, cell}) {
+        return this.find(`[data-cell="${cell}"][data-row="${row}"]`, this.gridContainer);
+    }
+
+    #start() {
+        let middleCell = Math.floor(this.gridCount) / 2;
+
+        this.#snake = this.#buildSnake(middleCell, middleCell);
+        this.#speed = +this.#form.speed.value;
+
+        this.direction = D.LEFT;
+
+        this.#setMessage('Welcome to Snake!');
+        this.#showEndGameButton();
+        this.#generateFood();
+
+        this.#processAction();
+        this.#process = setInterval(this.#processAction.bind(this), this.#speed);
+    }
+
+    #processAction() {
+        let { cell, row } = this.#noWallMode(this.#snake[0]);
+        let snakePartToShift = null;
+
+        switch(this.direction) {
+            case D.LEFT: {
+                snakePartToShift = {
+                    cell: cell - 1,
+                    row,
+                };
+            } break;
+            case D.RIGHT: {
+                snakePartToShift = {
+                    cell: cell + 1,
+                    row,
+                };
+            } break;
+            case D.UP: {
+                snakePartToShift = {
+                    cell,
+                    row: row - 1,
+                };
+            } break;
+            case D.DOWN: {
+                snakePartToShift = {
+                    cell,
+                    row: row + 1,
+                };
+            } break;
+        }
+
+        this.#snake.unshift(snakePartToShift);
+
+        this.#clear();
+        this.#update();
+    }
+
+    #update() {
+        for( let [index, snakePart] of this.#snake.entries()) {
+            let cellElement = this.#findByCoords(snakePart);
+
+            if(index === 0) {
+                cellElement.classList.add(Snake.snakeHeadCssClass, Snake.snakeCssClass);
+
+                this.#head = cellElement;
```

Comment:
What is the reason you need this ? We always can get access to this.#snake, don't we?